utils/fill_zero.go

```diff
@@ -24,88 +21,61 @@ func NewFillZeroCmd(trace bool) *FillZero {
 	return &FillZero{}
 }
 
-func (z *FillZero) Fill(ctx context.Context, logicalName string) error {
-	log.Printf("Start overwriting with zeros %s", logicalName)
-	partitionPath := logicalName // example /dev/sdb
-
-	// Buffer size (in bytes)
-	bufferSize := 4096 // to use 1MB = 1024 * 1024
-
+//noline:gomnd // the magic numbers here are fine
+func (z *FillZero) WipeDisk(ctx context.Context, path string) error {
+	fmt.Println("Starting zero-fill of", path)
 	// Write open
-	file, err := os.OpenFile(partitionPath, os.O_WRONLY, FileMode644)
+	file, err := os.OpenFile(path, os.O_WRONLY, 0)
 	if err != nil {
-		log.Println("Failed to open :"+logicalName, err)
+		log.Println(fmt.Errorf("%w", err))
 		return err
```

Comment:
I'll leave this up to @joelrebel but we should only really be handling errors once. Here we're doing 2 things with it, logging and returning. I would rather just return the error and leave logging up to the caller, maybe they care maybe they dont.